driver-core/src/main/com/mongodb/internal/event/EventListenerHelper.java

```diff
@@ -74,20 +73,14 @@ public static ServerMonitorListener getServerMonitorListener(final ServerSetting
         }
     }
 
-    public static ServerListener createServerListener(final ServerSettings serverSettings, final ServerListener additionalServerListener) {
```

Comment:
I'm a little confused what's going on here... First of all why did this method need to change to not need an additionalServerListener anymore? I couldn't really follow the changes you made to where this gets called.

I think I understand what this does - takes in a bunch of server listeners and creates a new one that broadcasts the messages to all them, but I don't understand why this would be necessary? I thought servers would have two listeners after this ticket (one that communicates with the application, and one that communicates with the cluster (for the description change events). Why are there more than that?

Reply:
Ah I think I get the second part now... The cluster uses this because it has multiple servers and wants to send the same message to all of them, right? I'm still confused about the first question though.

Reply:
The `additionalServerListener` is the one that is created by the `Cluster` implementations.  In this patch `Cluster` implementations no longer create a `ServerListener`.  Instead they create an instance of the new `ServerDescriptionChangedListener` interface, which is no longer combined with the ones from the application.  Instead, DefaultServer takes an instance of `ServerDescriptionChangedListener` separately in its constructor.

> The cluster uses this because it has multiple servers and wants to send the same message to all of them, right? 

Cluster doesn't send messages to Server.  It's the other way around.  Each Server instance sends a message to its containing Cluster.